control-plane-operator/controllers/hostedcontrolplane/registryoperator/reconcile.go

```diff
@@ -0,0 +1,467 @@
+			"token-minter",
+		}
+		c.Args = []string{
+			"--service-account-namespace",
+			workerNamespace,
+			"--service-account-name",
+			workerServiceAccount,
+			"--token-file",
+			path.Join(volumeMounts.Path(c.Name, volumeWebIdentityToken().Name), tokenFile),
+			"--kubeconfig",
+			path.Join(volumeMounts.Path(c.Name, volumeAdminKubeconfig().Name), kas.KubeconfigKey),
+		}
+		c.VolumeMounts = volumeMounts.ContainerMounts(c.Name)
+	}
+}
+
+/*
+Command: []string{"/usr/bin/control-plane-operator", "token-minter"},
+			Args: []string{
+				"--service-account-namespace=openshift-ingress-operator",
+				"--service-account-name=ingress-operator",
+				"--token-file=/var/run/secrets/openshift/serviceaccount/token",
+				"--kubeconfig=/etc/kubernetes/kubeconfig",
+			},
+			Resources: corev1.ResourceRequirements{
+				Requests: corev1.ResourceList{
+					corev1.ResourceCPU:    resource.MustParse("10m"),
+					corev1.ResourceMemory: resource.MustParse("10Mi"),
+				},
+			},
+			Image: params.TokenMinterImage,
+			VolumeMounts: []corev1.VolumeMount{
+				{Name: "serviceaccount-token", MountPath: "/var/run/secrets/openshift/serviceaccount"},
+				{Name: "admin-kubeconfig", MountPath: "/etc/kubernetes"},
+			},
+*/
+
+func volumeClientToken() *corev1.Volume {
+	return &corev1.Volume{
+		Name: "client-token",
+	}
+}
+
+func buildVolumeClientToken(v *corev1.Volume) {
+	v.EmptyDir = &corev1.EmptyDirVolumeSource{}
+}
+
+func volumeServingCert() *corev1.Volume {
+	return &corev1.Volume{
+		Name: "serving-cert",
+	}
+}
+
+func buildVolumeServingCert(v *corev1.Volume) {
+	v.Secret = &corev1.SecretVolumeSource{
+		SecretName: manifests.ImageRegistryOperatorServingCert("").Name,
+	}
+}
+
+func volumeAdminKubeconfig() *corev1.Volume {
+	return &corev1.Volume{
+		Name: "kubeconfig",
+	}
+}
+
+func buildVolumeAdminKubeconfig(v *corev1.Volume) {
+	v.Secret = &corev1.SecretVolumeSource{
+		SecretName: manifests.KASServiceKubeconfigSecret("").Name,
+	}
+}
+
+func volumeCABundle() *corev1.Volume {
+	return &corev1.Volume{
+		Name: "ca-bundle",
+	}
+}
+
+func buildVolumeCABundle(v *corev1.Volume) {
+	v.Secret = &corev1.SecretVolumeSource{
+		SecretName: manifests.RootCASecret("").Name,
+	}
+}
+
+func volumeWebIdentityToken() *corev1.Volume {
+	return &corev1.Volume{
+		Name: "web-identity-token",
+	}
+}
+
+func buildVolumeWebIdentityToken(v *corev1.Volume) {
+	v.EmptyDir = &corev1.EmptyDirVolumeSource{}
+}
+
+func ReconcileService(svc *corev1.Service) {
+	svc.Spec.ClusterIP = "None"
+	var port corev1.ServicePort
+	if len(svc.Spec.Ports) > 0 {
+		port = svc.Spec.Ports[0]
+	} else {
+		svc.Spec.Ports = []corev1.ServicePort{port}
+	}
+	port.Name = "metrics"
+	port.Port = int32(metricsPort)
+	port.Protocol = corev1.ProtocolTCP
+	port.TargetPort = intstr.FromInt(metricsPort)
+	svc.Spec.Ports[0] = port
+	svc.Spec.Selector = selectorLabels()
+}
+
+func ReconcileServiceMonitor(sm *prometheusoperatorv1.ServiceMonitor, clusterID string, metricsSet metrics.MetricsSet) {
```

Comment:
If we use a podmonitor, we do not need the service

Reply:
true, I'll convert